pkg/nsx/services/staticroute/staticroute.go

```diff
@@ -0,0 +1,232 @@
+	newNextHops := newStaticRoute.NextHops
+	if len(oldNextHops) != len(newNextHops) {
+		return false
+	}
+	oldHopsSet := sets.NewString()
+	for _, addr := range oldNextHops {
+		oldHopsSet.Insert(*addr.IpAddress)
+	}
+	for _, addr := range newNextHops {
+		if !oldHopsSet.Has(*addr.IpAddress) {
+			return false
+		}
+	}
+	return true
+}
+
+func (service *StaticRouteService) CreateorUpdateStaticRoute(obj *v1alpha1.StaticRoute) error {
+	nsxStaticRoute, err := service.buildStaticRoute(obj)
+	if err != nil {
+		return err
+	}
+
+	existingStaticRoute := service.StaticRouteStore.GetByKey(*nsxStaticRoute.Id)
+	if existingStaticRoute != nil && service.compareStaticRoute(existingStaticRoute, nsxStaticRoute) {
+		return nil
+	}
+	// TODO, update the value from 'default' to actual value
+	err = service.patch("default", "default", "default", nsxStaticRoute)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (service *StaticRouteService) buildStaticRoute(obj *v1alpha1.StaticRoute) (*model.StaticRoutes, error) {
+	sr := &model.StaticRoutes{}
+	sr.Network = &obj.Spec.Network
+	dis := int64(1)
+	for _, nh := range obj.Spec.NextHops {
+		nexthop := model.RouterNexthop{AdminDistance: &dis}
+		nexthop.IpAddress = &nh.IPAddress
+		sr.NextHops = append(sr.NextHops, nexthop)
+	}
+	sr.Id = (*string)(&obj.UID)
+	sr.Tags = service.buildBasicTags(obj)
+	return sr, nil
+}
+
+func (service *StaticRouteService) buildBasicTags(obj *v1alpha1.StaticRoute) []model.Tag {
+	tags := []model.Tag{
+		{
+			Scope: String(common.TagScopeCluster),
+			Tag:   String(service.NSXConfig.Cluster),
+		},
+		{
+			Scope: String(common.TagScopeNamespace),
+			Tag:   String(obj.ObjectMeta.Namespace),
+		},
+		// TODO: get namespace uid
+		{
+			Scope: String(common.TagScopeStaticRouteCRName),
+			Tag:   String(obj.ObjectMeta.Name),
+		},
+		{
+			Scope: String(common.TagScopeStaticRouteCRUID),
+			Tag:   String(string(obj.UID)),
+		},
+	}
+	return tags
+}
+
+func (service *StaticRouteService) patch(orgId string, projectId string, vpcId string, st *model.StaticRoutes) error {
+	err := service.NSXClient.StaticRouteClient.Patch(orgId, projectId, vpcId, *st.Id, *st)
+	if err != nil {
+		return err
+	}
+	err = service.StaticRouteStore.Add(*st)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (service *StaticRouteService) DeleteStaticRoute(orgId string, projectId string, vpcId string, uid string) error {
+	staticRouteClient := service.NSXClient.StaticRouteClient
+	staticroute := service.StaticRouteStore.GetByKey(uid)
+	if staticroute == nil {
+		return nil
+	}
+
+	if err := staticRouteClient.Delete(orgId, projectId, vpcId, *staticroute.Id); err != nil {
+		return err
+	}
+	if err := service.StaticRouteStore.Delete(*staticroute); err != nil {
+		return err
+	}
+
+	log.Info("successfully deleted NSX StaticRoute", "nsxStaticRoute", *staticroute.Id)
+	return nil
+}
+
+func queryStaticRoute(service *StaticRouteService, wg *sync.WaitGroup, fatalErrors chan error) {
```

Comment:
We can reuse function, refer [https://github.com/vmware-tanzu/nsx-operator/pull/160/files#diff-58e17097abee8cc07bbcbee5aa217c4e2e98799008f27d875636030d8f7d1f28R49](url)